.github/test-scripts/setup_pulp.sh

```diff
@@ -0,0 +1,117 @@
+
+echo "CONTENT_ORIGIN='http://$(hostname):8080'
+ANSIBLE_API_HOSTNAME='http://$(hostname):8080'
+ANSIBLE_CONTENT_HOSTNAME='http://$(hostname):8080/pulp/content'
+TOKEN_AUTH_DISABLED=True" >> settings/settings.py
+
+podman run --detach \
+           --publish 8080:80 \
+           --name pulp \
+           --volume "$(pwd)/settings":/etc/pulp \
+           --volume "$(pwd)/pulp_storage":/var/lib/pulp \
+           --volume "$(pwd)/pgsql":/var/lib/pgsql \
+           --volume "$(pwd)/containers":/var/lib/containers \
+           --device /dev/fuse \
+           pulp/pulp
+
+##############################################################################
+# Iteratively query the REST API until we get a JSON response (http code will
+# be 200) which will indicate the system is ready.
+##############################################################################
+
+set +e
+iter=0
+httpcode=""
+
+until [ "$httpcode" = "200" ]
+do
+  iter=$((iter+1))
+  if [ $iter == 20 ]; then
+    echo "Reached max iterations waiting for pulp container"
+    exit 1
+  fi
+  sleep 10
+  httpcode=$(curl -o /dev/null -s -w "%{http_code}\n" localhost:8080/pulp/api/v3/status/)
+done
+set -e
+
+##############################################################################
+# Set the 'admin' password and modify the pulp config with custom values.
+##############################################################################
+
+podman exec -it pulp bash -c 'pulpcore-manager reset-admin-password --password password'
+pulp config create --base-url http://localhost:8080 --username admin --password password
+
+##############################################################################
+# Create an image repo called 'testrepo'
+##############################################################################
+
+pulp container repository create --name testrepo
+
+##############################################################################
+# Log in to the local and remote registries and copy the images we wish to
+# mirror from remote to local repo. The remote login uses the password passed
+# in to this script ($1). It is a Github repo secret. We are using a remote
+# that requires authentication because we need a `builder` image (because
+# builder requires builder, yuck) and no signed builder image is available
+# in any open access repository, at this time.
+#
+# Setting XDG_RUNTIME_DIR is a workaround for some podman/skopeo issues.
+# See https://github.com/containers/skopeo/issues/942 for more info.
+##############################################################################
+
+export XDG_RUNTIME_DIR=/tmp/pulptests
+mkdir $XDG_RUNTIME_DIR
+
+skopeo login --username admin --password password localhost:8080 --tls-verify=false
+skopeo copy docker://registry.access.redhat.com/ubi9/ubi-micro:latest docker://localhost:8080/testrepo/ubi-micro --dest-tls-verify=false
+
+podman login --username '6340056|ansible-runner-gha' --password "$1" registry.redhat.io
```

Comment:
Thinking about this one a little more- the username should probably also be an arg provided by a GHA secret- not so much that it's a secret, but if someone wanted to run this from a fork, they'd have to provide their own token, which would have a different username. Since GHA doesn't really have a general concept of external "vars" that aren't secrets, a secret it'd have to be. :(

Reply:
Done.

Reply:
Seen as secrets here: https://github.com/Shrews/ansible-builder/runs/7254569323?check_suite_focus=true#step:6:149